src/core/eval.ml

```diff
@@ -315,8 +315,16 @@ and tree_walk : dtree -> ctxt -> stack -> (term * stack) option =
                     let stk = List.reconstruct left (body::args) right in
                     walk tr stk cursor vars_id id_vars
               end
+          | Kind
+          | Type
+          | Prod(_)
           | Meta(_, _) -> default ()
-          | _          -> assert false
+          | TRef(_)
+          | Appl(_)
+          | LLet(_)    -> assert false (* Should be reduced by [whnf_stk] *)
+          | Patt(_)
+          | TEnv(_)
+          | Wild       -> assert false (* Should not appear *)
```

Comment:
I would either put an `assert false` on each line, or only one at the end. In any case, I guess the comment should be placed above the first constructor for which it applies, don't you think?